params.go

```diff
@@ -45,6 +46,18 @@ type StringSliceParam struct {
 	value []string
 }
 
+type URLParam struct {
+	baseParam
+	def   string
```

Comment:
This is different than the rest of the types. I can't decide if I want:
```golang
	RedisURL      = params.URL("redis.url", &url.URL{Host: "localhost:6379", Scheme: "redis"})
```
or 
```golang
	RedisURL      = params.URL("redis.url", "redis://localhost:6379")
```

The issue with the first one is verbosity. The issue with the second one is this second `def` member, and the parsing of `def` into a url.URL, which can fail!